orb-slot-ctrl/src/lib.rs

```diff
@@ -174,6 +174,20 @@ impl TryFrom<u8> for RootFsStatus {
     }
 }
 
+impl TryFrom<&str> for RootFsStatus {
+    type Error = &'static str;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        match value.to_lowercase().as_str() {
+            "normal" | "0" => Ok(RootFsStatus::Normal),
+            "updateinprocess" | "updinprocess" | "1" => Ok(RootFsStatus::UpdateInProcess),
+            "updatedone" | "upddone" | "2" => Ok(RootFsStatus::UpdateDone),
+            "unbootable" | "3" => Ok(RootFsStatus::Unbootable),
+            _ => Err("Invalid status provided"),
```

Comment:
Either use a `()` as the Err variant, or create a struct and implement thiserr on it. IMO a str should not be used.